atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -78,4 +85,11 @@ protected boolean shouldStopRetrying(@SuppressWarnings("unused") int numTimesFai
             }
         }
     }
+
+    @Override
+    public void close() {
+        Preconditions.checkState(!isClosed, "Operations cannot be performed on closed TransactionManager.");
```

Comment:
You should be able to close an already closed txn manager, it should just be a no-op